src/poliastro/plotting.py

```diff
@@ -101,14 +102,19 @@ def set_attractor(self, orbit):
 
         self.ax.add_patch(mpl.patches.Circle((0, 0), radius, lw=0, color=color))
 
-    def plot(self, orbit, osculating=True, label=None):
+    def plot(self, orbit, osculating=True, label=None, propagate=False):
         """Plots state and osculating orbit in their plane.
 
         """
         # TODO: This function needs a refactoring
         if not self._frame:
             self.set_frame(*orbit.pqw())
 
+        # Propagates backwards of forwards to the last added orbit epoch.
+        if propagate:
+            if self._orbits:
+                orbit = orbit.propagate(self._orbits[-1].epoch - orbit.epoch)
```

Comment:
Well, there are two different approachs. We could store only the first epoch, or we could store only the epoch from the last orbit added. What do you think? In any case, you are right, only one epoch is needed.